lib/substep.js

```diff
@@ -0,0 +1,152 @@
+'use strict';
+
+const inherits      = require('util').inherits;
+const EventEmitter  = require('events').EventEmitter;
+const async         = require('async');
+
+const StepValue           = require('./step-value');
+const StepValueCollection = require('./step-value-collection');
+
+class Substep {
+  constructor(definitions, timeout) {
+    this.definitions = definitions;
+    this.stepTexts = [];
+    this.timeout = timeout;
+  }
+
+  runStep(text) {
+    this.stepTexts.push(text);
+    return this;
+  }
+
+  finish(callback) {
+    try {
+      this.runAll(callback);
+    } catch(e) {
+      callback(e);
+    }
+  }
+
+  runAll(callback) {
+    let steps = new MarrySteps(this.stepTexts, this.definitions.steps).marry();
+    steps.timeout = this.timeout;
```

Comment:
Should we pass this timeout to MarrySteps some how, and it adds the `timeout` to the steps?